tattle/model.py

```diff
@@ -55,19 +57,22 @@
 
 
 def determine_num_of_threads(thread_limit, num_of_items, per_page=1):
-
     requested_threads = int(math.ceil(num_of_items / float(per_page)))
     return min(thread_limit, requested_threads)
 
 
 def create_thread_pool(number_of_threads):
+    if number_of_threads == 0:
+        return ThreadPool()
     return ThreadPool(number_of_threads)
 
 
 def get_json(url, auth=None):
-
     response = requests.get(url, auth=auth)
-    return json.loads(response.text)
+    status_code = response.status_code
+    if 300 > status_code >= 200:
+        return json.loads(response.text)
```

Comment:
This might throw error even for 200 OK (with no content, response.text might not exist or be None)